multimodal_knowledge_retriever/streamlit/app.py

```diff
@@ -90,14 +104,22 @@ def main():
         st.markdown("**2. Set ingestion steps**")
         table_summaries = st.toggle("summarize Tables", value = True)
         text_summaries = st.toggle("summarize Text", value = False)
-        st.markdown("**3. Process your documents and create an in memory vector store**")
-        st.markdown("**Note:** Depending on the size and number of your documents, this could take several minutes")        
+        st.markdown("**3. Set retrieval steps**")
+        raw_image_retrieval = st.toggle("Answer over raw images", value = True)
+        st.caption("**Note** If selected the kit will use raw images to generate the answers, if not, image summaries will be used instead")
+        st.markdown("**4. Process your documents and create an in memory vector store**")
+        st.caption("**Note:** Depending on the size and number of your documents, this could take several minutes")        
         if st.button("Process"):
-            with st.spinner("Processing this could take a while..."):
-                st.session_state.qa_chain =  st.session_state.multimodal_retriever.st_ingest(
-                    docs, table_summaries, text_summaries
-                    )
-                st.toast("Vector DB successfully created!")
+            if docs:
+                with st.spinner("Processing this could take a while..."):
+                    st.session_state.qa_chain =  st.session_state.multimodal_retriever.st_ingest(
+                        docs, table_summaries, text_summaries, raw_image_retrieval
+                        )
+                    st.toast("Vector DB successfully created!")
+                    st.session_state.input_disabled=False
+                    st.rerun()
+            else:
+                st.error("You must provide at least one document", icon="🚨")
         st.markdown("**3. Ask questions about your data!**")
 
 if __name__ == "__main__":
```

Comment:
Which version of langchain community is needed for llms.sambanova?  :).   reqs.txt got me 0.0.38, but doesn't work.

Reply:
looks like it's working